lib/Conversion/AffineToPipeline/AffineToPipeline.cpp

```diff
@@ -337,6 +397,63 @@ LogicalResult AffineToPipeline::solveSchedulingProblem(
   return success();
 }
 
+LogicalResult AffineToPipeline::constantCSE() {
```

Comment:
The reason this is necessary is because materializing the `AffineApplyOp`s leaves lots of constants in the loop body.